tests/unit/multiclass/KNN_unittest.cc

```diff
@@ -8,98 +8,118 @@
 
 using namespace shogun;
 
-#ifdef HAVE_LAPACK
-void generate_knn_data(SGMatrix<float64_t>& feat, SGVector<float64_t>& lab,
-	   	int32_t num, int32_t classes, int32_t feats)
+class KNNTestData
```

Comment:
this test data class (and its usage in the unit tests) should be a sep. refactoring PR